.github/workflows/_upload.yml

```diff
@@ -0,0 +1,178 @@
+name: upload
```

Comment:
Please rename the composite action to `_binary_upload.yml`(or may be `binary_publish.yml`), as it is a very different from typical upload action we are doing.

Reply:
Makes sense, done.